strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/nuget/NugetArtifactController.java

```diff
@@ -118,15 +122,15 @@ public ResponseEntity deletePackage(@RequestHeader(name = "X-NuGet-ApiKey", requ
     {
         logger.info(String.format("Nuget delete request: storageId-[%s]; repositoryId-[%s]; packageId-[%s]", storageId, repositoryId, packageId));
 
-        String path = String.format("%s/%s/%s.nuspec", packageId, version, packageId);
-        
+        RepositoryPath path = repositoryPathResolver.resolve(storageId, repositoryId, String.format("%s/%s/%s.nuspec", packageId, version, packageId));;
         try
         {
-            nugetArtifactManagementService.delete(storageId, repositoryId, path, true);
-            path = String.format("%s/%s/%s.%s.nupkg", packageId, version, packageId,version);
-            nugetArtifactManagementService.delete(storageId, repositoryId, path, true);
-            path = String.format("%s/%s/%s.%s.nupkg.sha512", packageId, version, packageId,version);
-            nugetArtifactManagementService.delete(storageId, repositoryId, path, true);
+            //TODO: we should move associated files deletion into corresponding layout providers.
```

Comment:
+1 : I absolutely agree!

@sbespalov , or maybe @gokhankuyucak : Which one of you guys would like to do this? @sbespalov : Could you open a new task on Github for this?

Reply:
> @sbespalov : Could you open a new task on Github for this?
ok